### PR TITLE
[MRG, BUG] Fix consistency cast to upper for sex

### PR DESCRIPTION
**Pull request by alexrockhill**
PR Description
--------------

'm' and 'f' aren't currently recognized only 'M' and 'F', so fix by casting to upper.

Also report said `n` men and `n` women and these are gendered terms whereas male and female denote sex so probably best to use male/female.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by alexrockhill** [comment]
Ok, looks good to me now

**Review by agramfort — APPROVED** [review]

File: mne_bids/report.py [hunk 1]
```diff
@@ -201,10 +201,13 @@ def _summarize_participants_tsv(root, verbose=True):
     # summarize sex count statistics
     keys = ['M', 'F', 'n/a']
     p_sex = participants_tsv.get('sex')
+    # phrasing works for both sex and gender
+    p_gender = participants_tsv.get('gender')
     sexs = ['n/a']
-    if p_sex:
+    if p_sex or p_gender:
         # only summarize sex if it conforms to
         # mne-bids handedness
```
Comment (sappelhoff) [inline mne_bids/report.py:209]: I know it's not your change, but this comment is a bit cryptic :smile:

File: doc/whats_new.rst [hunk 2]
```diff
@@ -46,7 +46,7 @@ Requirements
 Bug fixes
 ^^^^^^^^^
 
-- ...
+- Fix m/f for male/female and l/r for left/right instead of only accpeting caps, by `Alex Rockhill`_ (:gh:`755`)
```
Comment (sappelhoff) [inline doc/whats_new.rst:49]: you have a typo there :) also, try to phrase these entries such that they make sense for somebody who will never look at your PR. For example:

```suggestion
- :func:`mne_bids.make_report` now (1) detects male/female sex and left/right handedness irrespective of letter case, (2) will parse a ``gender`` column if no ``sex`` column is found in ``participants.tsv``, and (3) reports sex as male/female instead of man/woman, by `Alex Rockhill`_ (:gh:`755`)
```

File: mne_bids/report.py [hunk 3]
```diff
@@ -201,10 +201,13 @@ def _summarize_participants_tsv(root, verbose=True):
     # summarize sex count statistics
     keys = ['M', 'F', 'n/a']
     p_sex = participants_tsv.get('sex')
+    # phrasing works for both sex and gender
+    p_gender = participants_tsv.get('gender')
```
Comment (sappelhoff) [inline mne_bids/report.py:205]: why did you add this "gender" column?

**Review by sappelhoff — COMMENTED** [review]
LGTM in general, but some minor comments.

File: doc/whats_new.rst (hunk 2 above)
Comment (alexrockhill) [reply doc/whats_new.rst:49]: Thanks :) My bad I modeled it off of Alex Gramfort's last bug comment haha

File: mne_bids/report.py (hunk 3 above)
Comment (alexrockhill) [reply mne_bids/report.py:205]: That's how the openneuro dataset in this PR https://github.com/mne-tools/mne-bids/pull/753 codes sex/gender

Comment (alexrockhill) [reply mne_bids/report.py:205]: I think it's not the only participants tsv that might have gender instead of sex

**Review by adam2392 — APPROVED** [review]
LGTM if CI works.

File: mne_bids/report.py (hunk 3 above)
Comment (sappelhoff) [reply mne_bids/report.py:205]: ok, fine with me :+1:

**Review by sappelhoff — APPROVED** [review]
LGTM once CIs pass :) thanks Alex

**Comment by agramfort** [comment]
thx @alexrockhill
